### PR TITLE
Cli update

### PR DESCRIPTION
**Pull request by douglas-raillard-arm**
Update the CLI:
* add a --verbose option
* generalize the use of logging module rather than print()
* unify configuration file handling with CLI by updating CLI parser default values

Based on: https://github.com/chrisdiamand/juniper-vpn-py/pull/2

### CODE REVIEW

File: junipervpn/tncc.py [hunk 1]
```diff
@@ -595,6 +593,8 @@ def process_cmd(self):
             dsid_value = args['Cookie']
 
 def main():
+    logging.basicConfig(level=logging.INFO)
```
Comment (chrisdiamand) [inline junipervpn/tncc.py:596]: Should we check the verbose flag here too? I know nothing uses it in tncc yet, but if someone did add a `logging.debug()` call they'd probably be confused about why it didn't show up with `--verbose`.

File: junipervpn/vpn.py [hunk 2]
```diff
@@ -313,14 +313,14 @@ def stop(self):
                 pass
 
 def main():
-    parser = argparse.ArgumentParser(conflict_handler='resolve')
+    parser = argparse.ArgumentParser(add_help=False)
```
Comment (chrisdiamand) [inline junipervpn/vpn.py:316]: Nice!

File: junipervpn/vpn.py [hunk 3]
```diff
@@ -65,6 +65,11 @@ def hotp(key):
     counter = int2beint64(int(time.time()) / 30)
     return dec(hmac.new(key, counter, hashlib.sha256).digest(), 6)
 
+class ActionError(Exception):
```
Comment (chrisdiamand) [inline junipervpn/vpn.py:67]: LGTM

File: junipervpn/vpn.py [hunk 4]
```diff
@@ -385,8 +385,7 @@ def main():
         args.action = shlex.split(args.action)
 
     if args.host == None or args.action == []:
-        print("--host and <action> are required parameters")
-        sys.exit(1)
+        parser.error("--host and <action> are required parameters")
```
Comment (chrisdiamand) [inline junipervpn/vpn.py:392]: LGTM

**Review by chrisdiamand — COMMENTED** [review]
I've looked at the easy ones so far:
```
4ed33d1 junipervpn.vpn: Remove conflicts in CLI parser
6540826 junipervpn.vpn: Add --verbose CLI option
fdfac62 junipervpn.vpn: Use parser.error() for CLI issues
4a4b365 junipervpn.vpn: Avoid backtrace on ctrl-C
```
I've added one simple comment to the `--verbose` one, but otherwise I think they could go straight in.

Would you prefer me to cherry-pick them, or do you want to move them to a separate PR? (I wish github allowed partial submission...)

I'll have a look at the last two once those have gone in :)

File: junipervpn/tncc.py (hunk 1 above)
Comment (douglas-raillard-arm) [reply junipervpn/tncc.py:596]: tncc as a standalone binary does not have a verbose option (it does not even use argparse). I have the feeling that it complies with some external specification, probably how openconnect calls this sort of "plugins".

When used through vpn.py, tncc.main() will not be called (as expected, this should only be called from the application, not any other library). This means the --verbose flag passed to vpn.py will enabled DEBUG level in tncc as well.

**Comment by douglas-raillard-arm** [comment]
> Would you prefer me to cherry-pick them, or do you want to move them to a separate PR? (I wish github allowed partial submission...)

I've reduced that PR to the list you shared and created another one for the 2 remaining commits:
https://github.com/chrisdiamand/juniper-vpn-py/pull/4

File: junipervpn/tncc.py (hunk 1 above)
Comment (chrisdiamand) [reply junipervpn/tncc.py:596]: Good point. tncc's arg parsing does looks pretty weird, let's leave it well alone then :D

**Review by chrisdiamand — APPROVED** [review]

**Comment by chrisdiamand** [comment]
Thanks!

**Pull request merged**